src/procedures/proc_functions.c

```
@@ -0,0 +1,141 @@
+ProcedureResult Proc_FunctionsFree
+(
+	ProcedureCtx *ctx
+) {
+	// clean up
+	if(ctx->privateData) {
+		ProcFunctionsPrivateData *pdata = ctx->privateData;
+		array_free(pdata->output);
+		rm_free(ctx->privateData);
+	}
+
+	return PROCEDURE_OK;
```

cppcheck

[error] Common realloc mistake

(memleakOnRealloc)